### PR TITLE
LPS-168723 useSidebarTask created

### PR DESCRIPTION
**Pull request by welitonpe**

### CODE REVIEW

**Review by rafaelfs98 — APPROVED** [review]
LGTM:fire:

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/Sidebar/TasksSidebar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+};
+
+const QuantityTaskBadge: React.FC<QuantityTaskBadgeProps> = ({
+	classStyle,
+	count,
+}) => {
+	return (
+		<div
+			className={classNames(
+				`align-items-center d-flex justify-content-center quantity-task-badge `,
```
Comment (samuelleolopes) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/Sidebar/TasksSidebar/index.tsx:48]: Put single quotes (' ')
```suggestion
				'align-items-center d-flex justify-content-center quantity-task-badge ',
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/Sidebar/SidebarItem.tsx [hunk 2]
```diff
@@ -47,8 +47,12 @@ const SidebarItem: React.FC<SidebarItemProps> = ({
 		>
 			<LinkComponent
 				className={classNames(
-					'cursor-pointer testray-sidebar-item',
-					{active},
+					'cursor-pointer  testray-sidebar-item',
```
Comment (samuelleolopes) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/Sidebar/SidebarItem.tsx:50]: Take unnecessary space

```suggestion
					'cursor-pointer testray-sidebar-item',
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/Sidebar/TasksSidebar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+const TaskSidebar: React.FC<TaskSidebarProps> = ({expanded}) => {
+	const {displayTask, setSelectedTask, subTasks, tasks} = useSidebarTask();
+
+	return (
+		<>
+			{tasks && (
+				<div className="task-sidebar">
+					<div
+						className={classNames(
+							'align-items-center d-flex  task-sidebar-title',
```
Comment (samuelleolopes) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/Sidebar/TasksSidebar/index.tsx:81]: Take unnecessary space

```suggestion
							'align-items-center d-flex task-sidebar-title',
```

**Review by samuelleolopes — COMMENTED** [review]
Check out the suggestions :)

**Comment by welitonpe** [comment]
ci:test:sf

**Comment by welitonpe** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/hooks/useSidebarTask.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+		subtasksResponse?.items,
+	]);
+
+	const showTask = () => {
+		selectedTask;
+
+		return tasks.find(({id}: {id: number}) => id === selectedTask);
+	};
+
+	const displayTask = showTask();
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/hooks/useSidebarTask.ts:61]: Do it

```suggestion
	const displayTask = tasks.find(({id}) => id === selectedTask);
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/hooks/useSidebarTask.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+	const tasks = useMemo(() => tasksResponse?.items || [], [
+		tasksResponse?.items,
+	]);
+
+	const subTasks = useMemo(() => subtasksResponse?.items || [], [
+		subtasksResponse?.items,
+	]);
+
+	const showTask = () => {
+		selectedTask;
```
[the remark this answers is not in the document]
Comment (kevenleone) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/hooks/useSidebarTask.ts:56]: Remove this function

**Review by kevenleone — CHANGES_REQUESTED** [review]
😄

**Review by kevenleone — APPROVED** [review]
LGTM 🔥

**Comment by kevenleone** [comment]
ci:forward

**Pull request closed without merging**